wagtail/admin/utils.py

```diff
@@ -20,3 +23,36 @@ def get_admin_base_url():
         admin_base_url = settings.BASE_URL
 
     return admin_base_url
+
+
+def get_block_usage():
+    """
+    Scans all StreamFields in the database and counts the number of times each block type has been used.
+
+    Returns:
+        A list of 2-tuples (block, usage_count) where 'block' is an instance of a Block and the usage_count
+        is the number of times that block has been used.
+    """
+    blocks = {}
+    counter = Counter()
+
+    for model in apps.get_models():
+        stream_fields = [
+            field
+            for field in model._meta.get_fields()
+            if isinstance(field, StreamField)
+        ]
+
+        if not stream_fields:
+            continue
+
+        for stream_values in model.objects.all().values_list(
```

Comment:
considering we are storing the data in a JSONField, could we not leverage that to run more performant queries?

Reply:
At CFPB @willbarton and I have been prototyping some code that makes use of Postgres-specific JSON functions and recursive queries to do something similar to this: filtering and annotating a PageQuerySet by the blocks it contains.

We haven't had time to release this as a standalone package yet, but sharing it here because the logic is very similar to what you would have to do to aggregate counts of block types: the file is [here](https://github.com/cfpb/consumerfinance.gov/blob/edba73a24e11572a411b8f0fe3863f018e691a63/cfgov/v1/query.py) and the meat of the query is [here](https://github.com/cfpb/consumerfinance.gov/blob/edba73a24e11572a411b8f0fe3863f018e691a63/cfgov/v1/query.py#L60-L94).

I don't know whether it's possible to implement something similar in every database backend; if/when the new block usage report gets into Wagtail core it'd be nice to architect it in such a way that we could potentially support more optimized queries for certain backends.